### PR TITLE
use torch.where over boolean masking

### PR DESCRIPTION
**Pull request by pmeier**
This solves the graph break for `torch.compile` that stems from pytorch/pytorch#114220. The fix proposed there is no fix for now, but rather just a pointer to setting a feature flag. This flag is supposed to become the default in the future. Still, using `torch.where` here adds no complexity and will be faster on CUDA as well.

cc @vfdev-5

### CODE REVIEW

**Comment by lezcano** [comment]
Also note that this will also generate better code even when https://github.com/pytorch/pytorch/issues/114220 is set to default.

**Review by lezcano — APPROVED** [review]

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @pmeier !
